media-placeholders/src/main/java/org/wordpress/aztec/placeholders/PlaceholderManager.kt

```diff
@@ -0,0 +1,338 @@
+            container.findViewWithTag<View>(uuid)?.let {
+                it.visibility = View.GONE
+                container.removeView(it)
+            }
+        }
+    }
+
+    /**
+     * Called before media is deleted. There is a delay between user deleting a media and when the media is actually is
+     * confirmed. That's why we first hide the media and we delete it when `onMediaDeleted` is actually called.
+     */
+    override fun beforeMediaDeleted(attrs: AztecAttributes) {
+        if (validateAttributes(attrs)) {
+            val uuid = attrs.getValue(UUID_ATTRIBUTE)
+            container.findViewWithTag<View>(uuid)?.let {
+                it.visibility = View.GONE
+            }
+        }
+    }
+
+    override fun canHandleTag(tag: String): Boolean {
+        return tag == HTML_TAG
+    }
+
+    /**
+     * This method handled a `placeholder` tag found in the HTML. It creates a placeholder and inserts a view over it.
+     */
+    override fun handleTag(opening: Boolean, tag: String, output: Editable, attributes: Attributes, nestingLevel: Int): Boolean {
+        if (opening) {
+            val type = attributes.getValue(TYPE_ATTRIBUTE)
+            val drawer = drawers[type] ?: return false
+            val drawable = buildPlaceholderDrawable(drawer)
+            val aztecAttributes = AztecAttributes(attributes)
+            aztecAttributes.setValue(UUID_ATTRIBUTE, UUID.randomUUID().toString())
+            val span = AztecPlaceholderSpan(
+                    context = aztecText.context,
+                    drawable = drawable,
+                    nestingLevel = nestingLevel,
+                    attributes = aztecAttributes,
+                    onMediaDeletedListener = this
+            )
+            val clickableSpan = AztecMediaClickableSpan(span)
+            val position = output.length
+            output.setSpan(span, position, position, Spanned.SPAN_MARK_MARK)
+            output.setSpan(clickableSpan, position, position, Spanned.SPAN_MARK_MARK)
+            output.append(Constants.IMG_CHAR)
+            output.setSpan(clickableSpan, position, output.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+            output.setSpan(span, position, output.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+            span.applyInlineStyleAttributes(output, position, output.length)
+
+            // The delay here is necessary because we don't know the view width before it's drawn.
+            launch {
+                delay(100)
+                // At this point we know the editor width so we need to update the drawable.
+                val editorWidth = aztecText.width
+                span.drawable?.setBounds(0, 0, editorWidth, drawer.getHeight(editorWidth))
+                aztecText.refreshText(false)
+                delay(100)
+                // Once the drawable and the placeholder are redrawn, we can finally insert the custom view over it.
+                insertInPosition(aztecAttributes, position)
+            }
```

Comment:
I think this can be made a bit more reliable using a `viewTreeObserver`. When I use this `EXAMPLE` editor text:

```
private val EXAMPLE =
        """
            <br>
            <placeholder id="126" type="image_with_caption" src="https://examplebloge.files.wordpress.com/2017/02/3def4804-d9b5-11e6-88e6-d7d8864392e0.png" caption="Happy times"/>
            <br>
            <placeholder id="0" type="image_with_caption" src="https://filesamples.com/samples/image/jpg/sample_640%C3%97426.jpg" caption="Test caption" />
            <br>
            <h1>Amanda was here</h1>
            <br>
            <placeholder id="1" type="image_with_caption" src="https://images.unsplash.com/photo-1608744882201-52a7f7f3dd60?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=736" caption="Test caption" />                 
            <br>
            <placeholder id="2" type="image_with_caption" src="https://examplebloge.files.wordpress.com/2017/02/3def4804-d9b5-11e6-88e6-d7d8864392e0.png" caption="Test caption" />
            <br>
        """.trimIndent()
```

 with the original code this is the result:

![original](https://user-images.githubusercontent.com/5810477/170143808-bded5ad2-318d-4a62-bfb2-ebf5e20c9df8.png)

but using the updated code below I get:

![view-tree-observer](https://user-images.githubusercontent.com/5810477/170143837-244b81b4-1d48-44af-acd5-b6df09ec593a.png)

```suggestion
            aztecText.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
                override fun onGlobalLayout() {
                    aztecText.viewTreeObserver.removeOnGlobalLayoutListener(this)
                    launch {
                        val editorWidth = aztecText.width
                        span.drawable?.setBounds(0, 0, editorWidth, drawer.getHeight(editorWidth))
                        aztecText.refreshText(false)
                        insertInPosition(aztecAttributes, position)
                    }
                }
            })
```

Reply:
good point, that's definitely more reliable 👍